uxarray/grid/grid.py

```diff
@@ -231,6 +231,24 @@ def from_dataset(
 
         return cls(grid_ds, source_grid_spec, source_dims_dict)
 
+    @classmethod
+    def from_geodataframe(
+        cls,
+        filename: str,
+        **kwargs,
+    ):
+        """Constructs a ``Grid`` object from a shapefile.
+
+        Parameters
+        ----------
+        filename : str
+           shapefile filename
+        """
+        source_grid_spec = "Shapefile"
```

Comment:
This should be generalized, either to `"Geometry File (format)"` or simply just `(format)`

Reply:
I gave it specific name based on extensions from filename - isn't that better?